### ISSUE TITLE
Why transaction support implemented only for SAConnection?

### ISSUE THREAD

**Issue by laz2** [aio-libs/aiopg#197]
I see only dependency from SA in [xid generation](https://github.com/aio-libs/aiopg/blob/9350678b8d369ea0e9ff734f358454e6d2992b7d/aiopg/sa/connection.py#L264). Other code uses basic aiopg and plain sql for work with transactions.

**Comment by asvetlov** [comment]
You can run raw SQL transactions by executing `BEGIN`/`COMMIT`/`ROLLBACK` statements

**Comment by laz2** [comment]
Yes. But without context manager and nested transaction it is not so comfortable. Do aiopg members have any reasons about this place of transactions code? May someone provide pull request with more generic  transaction support?

**Comment by asvetlov** [comment]
If you want to make a PR -- you a welcome

**Comment by vir-mir** [comment]
Hi!
I'm doing a transaction.
but my class tied to https://docs.python.org/3.5/library/typing.html
which appeared only in 3.4
Maybe add compatibility https://docs.python.org/3.5/library/typing.html
Or better to rewrite?

@asvetlov @jettify

**Comment by jettify** [comment]
What features do you need? It is possible to write async context managers in backward compatible way: https://github.com/aio-libs/aiopg/blob/master/aiopg/cursor.py#L380-L403 For what purpose do you need typing?

**Comment by vir-mir** [comment]
The fact that I have written it long time ago,only not cover the tests and docs
But it is written in python3.5 with [PEP484](https://www.python.org/dev/peps/pep-0484/)

But I looked at your code, very little corrections that-be to bring it to the [PEP3170](https://www.python.org/dev/peps/pep-3107/)

I will cook PR
